### PR TITLE
Content warnings

### PR DESCRIPTION
**Pull request by rabble**
I'm looking for feedback on style / code for this. The PR isn't done and shouldn't be merged yet as it only supports reports on notes. We need support for user reports/notes and also the labels that people themselves or others can place on content. 
Screenshot 2023-10-12 at 4 25 46 PM

### CODE REVIEW

File: Nos.xcodeproj/xcshareddata/xcschemes/Nos.xcscheme
Comment (mplorentz) [inline Nos.xcodeproj/xcshareddata/xcschemes/Nos.xcscheme:114]: This should be reverted before we merge as it slows down the queries quite a bit.

File: Nos/Assets/Localization/Localized.swift
Comment (mplorentz) [inline Nos/Assets/Localization/Localized.swift:257]: These codes are already localized in the `ReportCategory` type, although we don't have a good way to instantiate a `ReportCategory` from, say, the two letter code we would find in a report. Probably we need to write a static function on `ReportCategory` that takes a report `code` and recursively searches the `topLevelCategories` data for a matching category. Then you can get the localized name using the categories `displayName`.

File: Nos/Assets/SwiftGen Stencils/custom-xcassets.stencil
Comment (mplorentz) [inline Nos/Assets/SwiftGen Stencils/custom-xcassets.stencil:1]: Not sure what's up with this file. It should be `git rm`ed.

File: Nos/Models/Author+CoreDataClass.swift
Comment (mplorentz) [inline Nos/Models/Author+CoreDataClass.swift:198]: a Swiftier name would be `authoredReportsRequest()`

File: Nos/Models/Author+CoreDataClass.swift
Comment (mplorentz) [inline Nos/Models/Author+CoreDataClass.swift:230]: It's much faster to let the database filter the `kind` than to do it in memory here. You can add `AND kind = %@` to the predicate above (and then add EventKind.report.rawValue to the varargs).

**Review by mplorentz — COMMENTED** [review]
I didn't get through reviewing all of this but I'll leave the comments I have so far.

**Comment by rabble** [comment]
I think there is still a bug where it's hollowing out the buttons in a way that's not right.

**Review by mplorentz — APPROVED** [review]
I'm going to go ahead and merge this since it has been waiting for a review for a long time. There is more work to do on this feature, we have #650, #653, #654, #651, #482, #340 filed.

**Pull request merged**
